src/core/include/openvino/core/parallel.hpp

```diff
@@ -151,9 +165,16 @@ void parallel_nt(int nthr, const F& func) {
         func(0, 1);
         return;
     }
-
+    // We expect the number of threads here to be "nthr", so we need to disable dynamic behavior.
+    auto origin_dyn_val = omp_get_dynamic();
+    if (origin_dyn_val != 0) {
+        omp_set_dynamic(0);
```

Comment:
Why we have to disable nthreads adjustment here? Any details?

Reply:
This function `parallel_nt` is used with specified threads number. User expects the certain threads number and assign a routine for each of them, but OMP can reduce threads, which leads to unpredictable behavior.

Reply:
Thanks for the explanation. 
So I think we need to do the same changes inside parallel_nt_static?

Reply:
Done